### ISSUE TITLE
Crash when trying to use mTCP + PLIP.COM for a parallel port network.

### ISSUE THREAD

**Issue by yishayv** [86Box/86Box#3699]
**Describe the bug**
86Box crashes when trying to use DOS with mTCP + PLIP.COM for a parallel port network.
First offending commit:
c30d5d90b747f7d09829614289db62f3e312e0e4

**To Reproduce**
1. Create or modify any machine with the following network settings:
```INI
[Network]
net_01_card = plip
net_01_net_type = slirp

[Ports (COM & LPT)]
lpt1_device = plip
```  
2. Install FreeDOS (DOS 3.3 or newer work as well), mTCP and the PLIP.COM 
[86box.cfg.txt](https://github.com/86Box/86Box/files/12677382/86box.cfg.txt)
packet driver.
3. Load PLIP (If I don't specify an ethernet address it defaults to `00:00:00:00:00:00` - but it doesn't seem to matter anyway) with:
```
PLIP 0x65 7 0x378 04:FD:7A:11:ED:D9
```
4. Create a config file starting from `sample.cfg` and change the interrupt to 0x65:
```
packetint 0x65
```
5. `SET MTCPCFG=<PATH-OF-MTCP-CONFIG-FILE>`
6. `DHCP`

**Expected behavior**
DHCP gets a proper IP address and networking is available.

**Code**
```C
      263  static bool
      264  network_queue_full(netqueue_t *queue)
      265  {
  >   266      return ((queue->head + 1) & NET_QUEUE_LEN_MASK) == queue->tail;
      267  }
```

**Stack Trace**
```
$ gdb --tui --args build/src/86Box.exe --rompath 'C:\emu\86Box\roms' --vmpath '"C:\emu\86Box VMs\Debug"' '-O'
  (gdb) bt
#0  network_queue_full (queue=queue@entry=0x148) at C:/Git/86Box/src/network/network.c:266
#1  0x00007ff771388047 in network_queue_put (queue=0x148, data=0x220e1565a20 "\004z\021b", len=307) at C:/Git/86Box/src/network/network.c:286
#2  0x00007ff7713881e7 in network_tx (card=0x0, bufp=0x220e1565a00 "", len=307) at C:/Git/86Box/src/network/network.c:575
#3  0x00007ff771391f38 in plip_write_data (val=<optimized out>, priv=0x220e155a8c0) at C:/Git/86Box/src/network/net_plip.c:237
#4  0x00007ff771337467 in lpt_write (port=<optimized out>, val=1 '\001', priv=0x7ff773694140 <lpt_ports>) at C:/Git/86Box/src/lpt.c:114
#5  0x00007ff7714f33e8 in outb (port=<optimized out>, val=1 '\001') at C:/Git/86Box/src/io.c:356
#6  0x00007ff771226943 in opOUT_AL_DX (rmdat=<optimized out>) at C:/Git/86Box/src/cpu/x86_ops_io.h:115
#7  0x00007ff771170573 in exec386 (cycs=<optimized out>) at C:/Git/86Box/src/cpu/386.c:194
#8  0x00007ff7714e9085 in pc_run () at C:/Git/86Box/src/86box.c:1280
#9  0x00007ff7715064a1 in main_thread_fn () at C:/Git/86Box/src/qt/qt_main.cpp:119
#10 0x00007ff7727ac457 in std::__invoke_impl<void, void (*)()> (__f=<error reading variable>) at C:/Users/user/scoop/apps/msys2/2023-07-18/mingw64/include/c++/13.2.0/bits/invoke.h:60
#11 std::__invoke<void (*)()> (__fn=<error reading variable>) at C:/Users/user/scoop/apps/msys2/2023-07-18/mingw64/include/c++/13.2.0/bits/invoke.h:96
#12 std::thread::_Invoker<std::tuple<void (*)()> >::_M_invoke<0ull> (this=0x150) at C:/Users/user/scoop/apps/msys2/2023-07-18/mingw64/include/c++/13.2.0/bits/std_thread.h:292
#13 std::thread::_Invoker<std::tuple<void (*)()> >::operator() (this=0x150) at C:/Users/user/scoop/apps/msys2/2023-07-18/mingw64/include/c++/13.2.0/bits/std_thread.h:299
#14 std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)()> > >::_M_run (this=0x148) at C:/Users/user/scoop/apps/msys2/2023-07-18/mingw64/include/c++/13.2.0/bits/std_thread.h:244
#15 0x00007ff772850d4f in execute_native_thread_routine ()
#16 0x00007ff771c41fcb in pthread_create_wrapper ()
#17 0x00007fff60cbe634 in msvcrt!_beginthreadex () from C:\WINDOWS\System32\msvcrt.dll
#18 0x00007fff60cbe70c in msvcrt!_endthreadex () from C:\WINDOWS\System32\msvcrt.dll
#19 0x00007fff5fa0257d in KERNEL32!BaseThreadInitThunk () from C:\WINDOWS\System32\kernel32.dll
#20 0x00007fff6176aa68 in ntdll!RtlUserThreadStart () from C:\WINDOWS\SYSTEM32\ntdll.dll
#21 0x0000000000000000 in ?? ()
Backtrace stopped: previous frame inner to this frame (corrupt stack?)

(gdb) p queue
$3 = (netqueue_t *) 0x148
(gdb) p *queue
Cannot access memory at address 0x148
```

**Desktop (please complete the following information):**
 - OS: Windows 11
 - 86Box version: v4.0
 - Build information: Windows-64-b5000
 - Bisected with: local build - old dynarec, Mingw64 gcc Debug

**Additional context**
Thanks for continuing to improve this already amazing emulator :)

**Comment by raszpl** [comment]
I can also crash 86Box version: v4.0 using network in slightly different circumstances:
1. enabled SLIRP ne2000 
2. run ne2000 0x60
3. run Necromancers Dos Navigator
4. Press Alt+F1
5. select FTP-Server
6. select ftp.simtel.net

86Box will crash before making any network requests. I also tried 3Com 501 and that doesnt crash, but Etherlink II aka 503 again crashes.

**Comment by yishayv** [comment]
The 'card' member of the LPT instance wasn't initialized, which caused the issue.
I tried swapping the initialization order of the LPT and Network modules, and it seems to fix the problem. However, there's a question about potential side-effects from this change, since it is so high-up the code chain. I've created a pull request at https://github.com/yishayv/86Box/pull/1. Should I open it against the main project?
